lib/eventFactory.js

```diff
@@ -133,6 +136,92 @@ function getJobsFromPR(config) {
     return jobsToStart.filter(j => j.state === 'ENABLED' && !j.archived);
 }
 
+/**
+ * Implementation of queue prototype
+ * @method Queue
+ */
+function Queue() {
+    this.elements = [];
+}
+
+Queue.prototype.enqueue = e => {
+    this.elements.push(e);
+};
+
+Queue.prototype.isEmpty = () => {
+    return this.elements.length === 0;
+};
+
+Queue.prototype.dequeue = () => {
+    if (this.isEmpty()) return null;
+
+    return this.items.shift();
+};
+
+/**
+ * Filter workflow graph by bfs traversal wrt the subscribed job
+ * @method filterSubscribedJobs
+ * @param  {Object}    workflowGraph
+ * @return {Object}                                         Filtered workflow graph
+ */
+function filterSubscribedJobs(workflowGraph) {
```

Comment:
why is this needed, won't setting `startFrom` as `subscribed` take care of the logic ?

Reply:
What I inferred from the problem is that `~subscribed` is an additional flag that acts as a **selector** in addition to any other trigger that should be there as the `startFrom` value.
So in order to remove any ambiguity, I thought of pulling out this selection logic outside.
This can also help in implementing the label based subscription selector as was discussed, more clearly.